tests/protocols/test_protocols.py

```diff
@@ -0,0 +1,50 @@
+import tempfile
+
+import mypy.api
+import pytest
+from ophyd import sim
+
+from ophyd_async import protocols as bs_protocols
+
+
+def test_readable():
+    assert isinstance(sim.motor1, bs_protocols.AsyncReadable)
+    assert isinstance(sim.det1, bs_protocols.AsyncReadable)
+    assert not isinstance(sim.flyer1, bs_protocols.AsyncReadable)
+
+
+def test_pausable():
+    assert isinstance(sim.det1, bs_protocols.AsyncPausable)
```

Comment:
I'm concerned that these pass, since the ophyd (not async) sim objects are explicitly _not_ `AsyncReadable`, `AsyncPausable` etc.

The test cases should look something like this:
| Library used to write device | `AsyncReadable` | `Readable` |
|------------------------------|-----------------|------------|
| Original ophyd               | No              | Yes        |
| ophyd-async                  | Yes             | Yes        |

But we should figure out why these pass, ophyd objects are not async, so they shouldn't register as async

Reply:
They pass because `isinstance` checks on protocols are essentially `hasattr` checks

Reply:
Is `check_supports` more comprehensive?

Reply:
nope